src/Sarif/FileRegionsCache.cs

```diff
@@ -34,6 +44,9 @@ public FileRegionsCache(int capacity = DefaultCacheCapacity, IFileSystem fileSys
 
             // Build a cache for this data, with the load method it should use to add new entries
             _cache = new Cache<string, Tuple<string, NewLineIndex>>(BuildIndexForFile, capacity);
+
+            // Build a cache of rolling hashes (partial hash per line) for this data
+            _rollingHashesCache = new Cache<string, Dictionary<int, string>>(BuildRollingHashIndexForFile, capacity);
```

Comment:
@michaelcfanning I am creating a new cache, instead of extending the existing `_cache`, to avoid lugging around an additional dictionary in situations where the user does not want to compute rolling hashes.